### PR TITLE
More cleanup, this time focused on boost

### LINKED ISSUE #128: housekeeping: Check that all boost lib are used
as a follow up to #126 , we might have many boost libraries that are detected and linked with that are not used.

### PR DESCRIPTION
**Pull request by greenscientist**
We had several boost libraries defined and not used, so let's remove them. 

This build up on PR #130. Remove some unused code that were referencing some boost libs.

### CODE REVIEW

**Comment by tahini** [comment]
there's an include of gtfs_fetcher in the benchmark.

**Comment by greenscientist** [comment]
ok, I got confused with all the parallel clean up commit :)

File: tests/benchmark_csa/benchmark_CSA_test.cpp [hunk 1]
```diff
@@ -3,13 +3,12 @@
 //Added for the json-example
 #define BOOST_SPIRIT_THREADSAFE
 
+#include <fstream>
 #include <boost/uuid/uuid.hpp>
 #include <boost/uuid/string_generator.hpp>
 
 #include "calculator.hpp"
 #include "cache_fetcher.hpp"
-#include "gtfs_fetcher.hpp" // To initialize calculator object
```
Comment (tahini) [inline tests/benchmark_csa/benchmark_CSA_test.cpp:11]: These lines should be part of your first commit. If we ever bisect around here, there will be broken commits fwiw...

**Review by tahini — APPROVED** [review]
Less dependencies is always good!

File: tests/benchmark_csa/benchmark_CSA_test.cpp (hunk 1 above)
Comment (greenscientist) [reply tests/benchmark_csa/benchmark_CSA_test.cpp:11]: fixing this...

**Comment by greenscientist** [comment]
Should be ok now

**Comment by tahini** [comment]
Thanks, let's merge!

**Pull request merged**